File: app/src/main/java/md/fusionworks/lifehack/di/component/ActivityComponent.java
package md.fusionworks.lifehack.di.component;

import dagger.Component;
import md.fusionworks.lifehack.di.module.ActivityModule;
import md.fusionworks.lifehack.di.scope.PerActivity;
import md.fusionworks.lifehack.ui.activity.BaseActivity;
import md.fusionworks.lifehack.ui.activity.NavigationDrawerActivity;

/**
 * Created by ungvas on 10/21/15.
 */
@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(BaseActivity baseActivity);

    void inject(NavigationDrawerActivity navigationDrawerActivity);
}

File: app/src/main/java/md/fusionworks/lifehack/di/ActivityComponent.kt
package md.fusionworks.lifehack.launch

import dagger.Component
import md.fusionworks.lifehack.application.AppComponent
import md.fusionworks.lifehack.di.PerActivity
import md.fusionworks.lifehack.main.MainActivity
import md.fusionworks.lifehack.taxi.TaxiActivity

/**
 * Created by ungvas on 3/21/16.
 */
@PerActivity
@Component(dependencies = arrayOf(AppComponent::class))
interface ActivityComponent {

  fun inject(launchActivity: LaunchActivity)

  fun inject(mainActivity: MainActivity)

  fun inject(taxiActivity: TaxiActivity)
}